sdk/core/core-amqp/src/retry.ts

```diff
@@ -219,21 +219,23 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
         i,
         err
       );
-      let targetDelayInMs = config.retryOptions.retryDelayInMs;
-      if (config.retryOptions.mode === RetryMode.Exponential) {
-        let incrementDelta = Math.pow(2, i) - 1;
-        const boundedRandDelta =
-          config.retryOptions.retryDelayInMs * 0.8 +
-          Math.floor(
-            Math.random() *
-              (config.retryOptions.retryDelayInMs * 1.2 - config.retryOptions.retryDelayInMs * 0.8)
-          );
-        incrementDelta *= boundedRandDelta;
-
-        targetDelayInMs = Math.min(incrementDelta, config.retryOptions.maxRetryDelayInMs);
-      }
 
       if (lastError && lastError.retryable && totalNumberOfAttempts > i) {
+        let targetDelayInMs = config.retryOptions.retryDelayInMs;
+        if (config.retryOptions.mode === RetryMode.Exponential) {
+          let incrementDelta = Math.pow(2, i) - 1;
+          const boundedRandDelta =
+            config.retryOptions.retryDelayInMs * 0.8 +
+            Math.floor(
+              Math.random() *
+                (config.retryOptions.retryDelayInMs * 1.2 -
+                  config.retryOptions.retryDelayInMs * 0.8)
+            );
+          incrementDelta *= boundedRandDelta;
```

Comment:
NIT: make `incrementDelta` a constant to make it easier to reason about the logic. Also, `retry` looks too big to me, should the logic for calculating `targetDelayInMs` for the exponential case go into its own function?
```suggestion
          const boundedRandDelta =
            config.retryOptions.retryDelayInMs * 0.8 +
            Math.floor(
              Math.random() *
                (config.retryOptions.retryDelayInMs * 1.2 -
                  config.retryOptions.retryDelayInMs * 0.8)
            );
          const incrementDelta = (Math.pow(2, i) - 1) * boundedRandDelta;
```

Reply:
Good point!